src/blob/index/bptree/core.rs

```diff
@@ -153,29 +153,61 @@ impl BPTreeFileIndex {
         if read_buf_size != buf_size {
             Err(anyhow!("Can't read entire leaf node"))
         } else {
-            self.read_header_buf(&buf[..buf_size], key, self.header.record_header_size)
-                .map(|opt| opt.map(|(header, _)| header))
+            if let Some((record_header, offset)) =
+                self.read_header_buf(&buf[..buf_size], key, self.header.record_header_size)?
+            {
+                let leftmost_header = self.get_leftmost(
+                    &buf[..buf_size],
+                    key,
+                    offset as usize,
+                    record_header,
+                    self.header.record_header_size,
+                )?;
+                Ok(Some(leftmost_header))
+            } else {
+                Ok(None)
+            }
         }
     }
 
+    fn get_leftmost(
+        &self,
+        raw_headers_buf: &[u8],
+        key: &[u8],
+        mut offset: usize,
+        mut prev_header: RecordHeader, // it's expected, that this header is from raw_headers_buf[offset..]
+        record_header_size: usize,
+    ) -> Result<RecordHeader> {
+        while offset > 0 {
```

Comment:
Is it possible that there is no space before the current header for a one more header?
(Example: header size is 16 bytes and offset is 8 bytes)

Reply:
No, it's not possible, because it's guaranteed that every leaf node starts from header (and even more: it's guaranteed, that there is no situation, that first header in node has the same key, as previous header) and we read first 4kb of node in `raw_headers_buf`